python/main.py

```diff
@@ -98,8 +111,66 @@ async def get_image(image_name):
 class Item(BaseModel):
     name: str
     category: str
+    image_name: str
 
 
-def insert_item(item: Item):
+def insert_item(item: Item, image_name: str):
     # STEP 4-1: add an implementation to store an item
-    pass
+    try:
+        if not JSON_DB.exists():
+            with open(JSON_DB, "w", encoding="utf-8") as f:
+                json.dump({"items": []}, f, indent=2)
+
+        with open(JSON_DB, "r+", encoding="utf-8") as f:
+            content = f.read().strip()
+            data = json.loads(content) if content else {"items": []}
+            logger.info("Succeeded to open json file")
+
+            if "items" not in data:
+                data["items"] = []
+
+            existing_item = next((i for i in data["items"] if i["name"] == item.name), None)
+
+            if existing_item:
+                logger.info(f"Item already exists, updating image_name: {existing_item}")
+                existing_item["image_name"] = image_name
+            else:
+                new_item = item.dict()
+                new_item["image_name"] = image_name
+                data["items"].append(new_item)
+                logger.info(f"New item inserted: {new_item}")
+
+            f.seek(0)
+            json.dump(data, f, indent=2, ensure_ascii=False)
+            f.truncate()
+
+    except Exception as e:
+        logger.error(f"Failed to save item: {e}")
+        raise HTTPException(status_code=500, detail="Failed to save item")
+
+
+
+@app.get("/items")
+@app.get("/items/{item_id}")
+def get_items(item_id: int = None):
+    try:
+        if not JSON_DB.exists():
+            return {"items": []}
```

Comment:
We should return an None here if `item_id` is not `None`.

Alternatively to keep things easy to read, we can separate handlers for `/items` and `/items/{item_id}` in separate functions, which would be my suggestion.